### PR TITLE
Hide most-relevant sorting option when there is no text query

### PR DESCRIPTION
**Pull request by mariojsnunes**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Hides MostRelevant option when there is no search text.
Fixes sorting option displayed value (now always displays the label instead of the value - "Most Relevant" instead of "MostRelevant")
Improved typescript usage and aligned patterns of Howto/Research/Questions search

## Git Issues

Closes #

### CODE REVIEW

**Comment by codisart** [comment]
In the state of this PR, if the user use the url `/research?sort=MostRelevant` without the q param, the select display `Most Relevant` but the option does not exists in the select options. Maybe not a bug but I thought I should notify it.

File: src/pages/Howto/Content/HowtoList/HowtoSortOptions.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-HowtosSortOptions.set('LatestUpdated', 'Latest Updated')
-HowtosSortOptions.set('MostUseful', 'Most Useful')
-HowtosSortOptions.set('MostDownloads', 'Most Downloads')
+export const HowtoSortOptions = new Map<HowtoSortOption, string>()
+HowtoSortOptions.set('MostRelevant', 'Most Relevant')
+HowtoSortOptions.set('Newest', 'Newest')
+HowtoSortOptions.set('MostComments', 'Most Comments')
+HowtoSortOptions.set('LatestUpdated', 'Latest Updated')
+HowtoSortOptions.set('MostUseful', 'Most Useful')
+HowtoSortOptions.set('MostDownloads', 'Most Downloads')
```
Comment (codisart) [inline src/pages/Howto/Content/HowtoList/HowtoSortOptions.ts:15]: suggestion: I think we could have the Options behavior encapsulated in this file

```typescript
const BaseOptions = new Map<HowtoSortOption, string>()
BaseOptions.set('Newest', 'Newest')
BaseOptions.set('MostComments', 'Most Comments')
BaseOptions.set('LatestUpdated', 'Latest Updated')
BaseOptions.set('MostUseful', 'Most Useful')
BaseOptions.set('MostDownloads', 'Most Downloads')

const QueryParamOptions = new Map<HowtoSortOption, string>(BaseOptions)
QueryParamOptions.set('MostRelevant', 'Most Relevant')

const toArray = (hasQueryParam: boolean) => {
  const options = hasQueryParam ? QueryParamOptions : BaseOptions
  return Array.from(options, ([value, label]) => ({label: label, value: value }))
}

export const HowtoSortOptions = {
  get: (key: HowtoSortOption) => QueryParamOptions.get(key) ?? '',
  toArray
}
```

Having this wrapper allows to use the `HowtoSortOptions` like this 

```typescript
 <Select
            options={HowtoSortOptions.toArray(!!q)}
            placeholder={listing.sort}
            value={{
              label: HowtoSortOptions.get(sort),
              value: sort,
            }}
            onChange={(sortBy) =>
              updateFilter(HowtosSearchParams.sort, sortBy.value)
            }
            ```

File: src/pages/Question/QuestionSortOptions.ts [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
-}
+export type QuestionSortOption =
+  | 'MostRelevant'
+  | 'Newest'
+  | 'LatestUpdated'
+  | 'LatestComments'
+  | 'Comments'
+  | 'LeastComments'
+
+export const QuestionSortOptions = new Map<QuestionSortOption, string>()
```
Comment (codisart) [inline src/pages/Question/QuestionSortOptions.ts:9]: it seems that `src/pages/Question/QuestionListing.tsx` is still using the old enum

**Comment by mariojsnunes** [comment]
thank you @codisart!
I'll address your comments at the original PR https://github.com/ONEARMY/community-platform/pull/3478

**Pull request closed without merging**